pkgs/top-level/all-packages.nix

```diff
@@ -157,11 +40,7 @@ let
   #
   # The result is `pkgs' where all the derivations depending on `foo'
   # will use the new version.
-  overridePackages = f:
-    let
-      newpkgs = pkgsFun newpkgs overrides;
-      overrides = mkOverrides pkgs (f newpkgs pkgs);
-    in newpkgs;
+  overridePackages = f: pkgsWithOverrides f;
```

Comment:
I am not sure what should be done with this `overridePackages` function.  I personally have no uses of it, and I am sure we can discuss that, or just remove it if not used.

I did not went for using the unfix attribute for one simple reason.  The current version of `overridePackages` does not contain any of the modifications made in `~/.nixpkgs.config.nix`.  Thus, adding such a thing would probably be bad, as one might rely on such feature to provide its own set of overridden packages inside a repository.